crates/circuit/src/circuit_drawer.rs

```diff
@@ -0,0 +1,916 @@
+                                    None =>  match subnode_min_clbit {
+                                        Some(val) => val + total_qubits,
+                                        None => match subnode_max_clbit {
+                                            Some(val) => val + total_qubits,
+                                            None => continue, // No qubits or clbits, skip this node
+                                        },                                                
+                                    }, // No qubits or clbits, skip this node
+                                },
+                            };
+                            let subnode_max = match subnode_max_clbit {
+                                Some(val) => val + total_qubits,
+                                None => match subnode_min_clbit {
+                                    Some(val) => val + total_qubits,
+                                    None =>  match subnode_max_qubit {
+                                        Some(val) => val,
+                                        None => match subnode_min_qubit {
+                                            Some(val) => val,
+                                            None => continue, // No qubits or clbits, skip this node
+                                        },                                                
+                                    }, // No qubits or clbits, skip this node
+                                },
+                            };
+                            if (subnode_min <= node_min && subnode_max >= node_min) || (subnode_min <= node_max && subnode_max >= node_max) {
+                                overlap = true;
+                                break;
+                            }
+                        }
+                    }
+
+                    if overlap {
+                        sublayers.push(vec![node_index]);
+                    } else {
+                        sublayer.push(node_index);
+                    }
+                }
+            }
+            
+            let mut ct = 0;
+            for j in sublayers {
+                if j.is_empty() {
+                    continue;
+                } else {
+                    final_layers.push(j);
+                    ct += 1;
+                }
+            }
+        }
+
+
+        let mut packedin_layers: Vec<Vec<&PackedInstruction>> = Vec::new();
+
+        for (id,layer) in final_layers.iter().enumerate() {
+            let mut packedin_layer: Vec<&PackedInstruction> = Vec::new();
+            for nodeind in layer {
+                if let NodeType::Operation(instruction) = &binding.dag()[*nodeind] {
+                    packedin_layer.push(instruction);
+                } 
+            }
+            // println!("Layer {}", id);
+            self.build_layer(packedin_layer.clone());
+        }
+        
+    }
+}
+
+
+pub fn circuit_draw(dag_circ: &DAGCircuit) {
```

Comment:
With the recommendation above, this function can look something like this: 
```rust
pub fn circuit_draw(circuit_data: &CircuitData) -> PyResult<()> {

    let mut output = String::new();

    // Create a circuit representation
    let mut circuit_rep = circuit_rep::new(DAGCircuit::from_circuit_data(circuit_data, false, None, None, None, None)?);
    circuit_rep.set_qubit_name();
    circuit_rep.build_layers();
    output.push_str(&circuit_rep.circuit_string());
    // Print the circuit representation
    println!("{}", output);
    Ok(())
    ```
    The main reason for this change being making it easier for callers with `CircuitData` to use this function without needing to convert to `DAGCircuit` first.